### PR TITLE
5143-Add Optimize Queries page with query analysis help

### PR DESCRIPTION
**Pull request by jstirnaman**
- Closes Client library query traces: Python https://github.com/influxdata/docs-v2/issues/5143
- Dedicated and Clustered examples for enabling query tracing and extracting headers
- System.queries table
- Explain and Analyze
- For now, skip tests for sample Flight responses until we add in code samples.

**Don't merge** until Python client library allows for setting query request headers (https://github.com/InfluxCommunity/influxdb3-python/pull/64)

### CODE REVIEW

**Comment by jstirnaman** [comment]
I'd appreciate you fact-checking me on `optimize-queries.md` for Dedicated, Clustered, and Serverless.

File: content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+<!-- TOC -->
+
+- [Troubleshoot query performance](#troubleshoot-query-performance)
+  - [EXPLAIN and ANALYZE](#explain-and-analyze)
+  - [Enable trace logging](#enable-trace-logging)
+    - [Extract a trace ID](#extract-a-trace-id)
+  - [Retrieve system information](#retrieve-system-information)
+    - [Retrieve query instance information](#retrieve-query-instance-information)
+
+<!-- /TOC -->
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md:31]:
```suggestion
Use the following tools to help you identify performance bottlenecks and troubleshoot problems in queries:

<!-- TOC -->

- [EXPLAIN and ANALYZE](#explain-and-analyze)
- [Enable trace logging](#enable-trace-logging)
  - [Extract a trace ID](#extract-a-trace-id)
- [Retrieve system information](#retrieve-system-information)
  - [Retrieve query instance information](#retrieve-query-instance-information)

<!-- /TOC -->
```

File: content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+
+### EXPLAIN and ANALYZE
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md:33]:
```suggestion
## EXPLAIN and ANALYZE
```

File: content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+
+To view the query engine's execution plan and metrics for an SQL query, prepend [`EXPLAIN`](/influxdb/cloud-dedicated/reference/sql/explain/) or [`EXPLAIN ANALYZE`](/influxdb/cloud-dedicated/reference/sql/explain/#explain-analyze) to the query.
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md:35]: This is true for both SQL and InfluxQL.

```suggestion
To view the query engine's execution plan and metrics for an SQL or InfluxQL query, prepend [`EXPLAIN`](/influxdb/cloud-dedicated/reference/sql/explain/) or [`EXPLAIN ANALYZE`](/influxdb/cloud-dedicated/reference/sql/explain/#explain-analyze) to the query.
```

File: content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+explain_and_analyze()
+```
+{{% /code-placeholders %}}
+
+Replace the following:
+
+- {{% code-placeholder-key %}}`DATABASE_NAME`{{% /code-placeholder-key %}}: your {{% product-name %}} database
+- {{% code-placeholder-key %}}`DATABASE_TOKEN`{{% /code-placeholder-key %}}: a [database token](/influxdb/cloud-dedicated/admin/tokens/) with sufficient permissions to the specified database
+
+The output is similar to the following:
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md:92]: There may be differences in the output between SQL and InfluxQL. Maybe some tabs here?

File: content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+|:------------------|:---------------------------------------------------------------------------------------------------------------------------------------------|
+| logical_plan      | Sort: home.time ASC NULLS LAST                                                                                                               |
+|                   |   TableScan: home projection=[co, hum, room, sensor, temp, time], full_filters=[home.time >= TimestampNanosecond(1688491380936276013, None)] |
+| physical_plan     | SortExec: expr=[time@5 ASC NULLS LAST]                                                                                                       |
+|                   |   EmptyExec: produce_one_row=false                                                                                                           |
+| Plan with Metrics | SortExec: expr=[time@5 ASC NULLS LAST], metrics=[output_rows=0, elapsed_compute=1ns, spill_count=0, spilled_bytes=0]                         |
+|                   |   EmptyExec: produce_one_row=false, metrics=[]
+```
+
+### Enable trace logging
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md:105]:
```suggestion
## Enable trace logging
```

File: content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md [hunk 6]
```diff
@@ -0,0 +1,410 @@
...
+
+Replace the following:
+
+- {{% code-placeholder-key %}}`DATABASE_NAME`{{% /code-placeholder-key %}}: your {{% product-name %}} database
+- {{% code-placeholder-key %}}`DATABASE_TOKEN`{{% /code-placeholder-key %}}: a [database token](/influxdb/cloud-dedicated/admin/tokens/) with sufficient permissions to the specified database
+- {{% code-placeholder-key %}}`APP_REQUEST_ID`{{% /code-placeholder-key %}}: a unique identifier you provide to associate with the trace logging.
+
+With tracing enabled, the response contains the `trace-id` header with the 128-bit unique _trace ID_ generated by the tracing system.
+
+#### Extract a trace ID
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md:223]:
```suggestion
### Extract a trace ID
```

File: content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md [hunk 7]
```diff
@@ -0,0 +1,410 @@
...
+
+  # Access the trace ID that the the middleware collected.
+  assert res.traces[0]
+  print(f"Received generated trace-id: {res.traces[0]}")
+
+query_with_middleware()
+```
+{{% /code-placeholders %}}
+
+### Retrieve system information
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md:300]:
```suggestion
## Retrieve system information
```

File: content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md [hunk 8]
```diff
@@ -0,0 +1,410 @@
...
+
+In addition to the SQL standard `information_schema`, {{% product-name %}} contains _system_ tables that provide access to
+InfluxDB-specific information.
+The information in each system table is scoped to the namespace you're querying;
+you can only retrieve system information for that particular instance.
+
+#### Retrieve query instance information
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md:307]:
```suggestion
### Retrieve query instance information
```

File: content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md [hunk 9]
```diff
@@ -0,0 +1,410 @@
...
+
+To get information about queries you've run on the current instance, use SQL to query the [`system.queries` table](/influxdb/cloud-dedicated/reference/internals/system-tables/#systemqueries-measurement).
+`system.queries` contains information about queries run against the IOx instance that is currently handling queries.
+If you [enabled trace logging for the query](#enable-trace-logging-for-a-query), the `trace-id` appears in the `system.queries.trace_id` column for the query run.
+
+The `system.queries` table is an InfluxDB v3 **debug feature**.
+To enable the feature and query `system.queries`, include an `"iox-debug"` header set to `"true"` and use SQL to query the table
```
Comment (sanderson) [inline content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md:314]:
```suggestion
To enable the feature and query `system.queries`, include an `"iox-debug"` header set to `"true"` and use SQL to query the table.
```

File: content/influxdb/cloud-serverless/query-data/execute-queries/optimize-queries.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+
+Use the following tools to help you identify performance bottlenecks and troubleshoot problems in queries:
+
+<!-- TOC -->
+
+- [Troubleshoot query performance](#troubleshoot-query-performance)
+  - [EXPLAIN and ANALYZE](#explain-and-analyze)
+  - [Enable trace logging](#enable-trace-logging)
+
+<!-- /TOC -->
```
Comment (sanderson) [inline content/influxdb/cloud-serverless/query-data/execute-queries/optimize-queries.md:28]:
```suggestion
Use the following tools to help you identify performance bottlenecks and troubleshoot problems in queries:

<!-- TOC -->

- [EXPLAIN and ANALYZE](#explain-and-analyze)
- [Enable trace logging](#enable-trace-logging)

<!-- /TOC -->
```

File: content/influxdb/cloud-serverless/query-data/execute-queries/optimize-queries.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+
+### EXPLAIN and ANALYZE
```
Comment (sanderson) [inline content/influxdb/cloud-serverless/query-data/execute-queries/optimize-queries.md:30]:
```suggestion
## EXPLAIN and ANALYZE
```

File: content/influxdb/cloud-serverless/query-data/execute-queries/optimize-queries.md [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+
+To view the query engine's execution plan and metrics for an SQL query, prepend [`EXPLAIN`](/influxdb/cloud-serverless/reference/sql/explain/) or [`EXPLAIN ANALYZE`](/influxdb/cloud-serverless/reference/sql/explain/#explain-analyze) to the query.
```
Comment (sanderson) [inline content/influxdb/cloud-serverless/query-data/execute-queries/optimize-queries.md:32]:
```suggestion
To view the query engine's execution plan and metrics for an SQL or InfluxQL query, prepend [`EXPLAIN`](/influxdb/cloud-serverless/reference/sql/explain/) or [`EXPLAIN ANALYZE`](/influxdb/cloud-serverless/reference/sql/explain/#explain-analyze) to the query.
```

File: content/influxdb/cloud-serverless/query-data/execute-queries/optimize-queries.md [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+|:------------------|:---------------------------------------------------------------------------------------------------------------------------------------------|
+| logical_plan      | Sort: home.time ASC NULLS LAST                                                                                                               |
+|                   |   TableScan: home projection=[co, hum, room, sensor, temp, time], full_filters=[home.time >= TimestampNanosecond(1688491380936276013, None)] |
+| physical_plan     | SortExec: expr=[time@5 ASC NULLS LAST]                                                                                                       |
+|                   |   EmptyExec: produce_one_row=false                                                                                                           |
+| Plan with Metrics | SortExec: expr=[time@5 ASC NULLS LAST], metrics=[output_rows=0, elapsed_compute=1ns, spill_count=0, spilled_bytes=0]                         |
+|                   |   EmptyExec: produce_one_row=false, metrics=[]
+```
+
+### Enable trace logging
```
Comment (sanderson) [inline content/influxdb/cloud-serverless/query-data/execute-queries/optimize-queries.md:102]:
```suggestion
## Enable trace logging
```

File: content/influxdb/clustered/query-data/execute-queries/optimize-queries.md [hunk 14]
```diff
@@ -0,0 +1,410 @@
...
+<!-- TOC -->
+
+- [Troubleshoot query performance](#troubleshoot-query-performance)
+  - [EXPLAIN and ANALYZE](#explain-and-analyze)
+  - [Enable trace logging](#enable-trace-logging)
+    - [Extract a trace ID](#extract-a-trace-id)
+  - [Retrieve system information](#retrieve-system-information)
+    - [Retrieve query instance information](#retrieve-query-instance-information)
+
+<!-- /TOC -->
```
Comment (sanderson) [inline content/influxdb/clustered/query-data/execute-queries/optimize-queries.md:31]:
```suggestion
Use the following tools to help you identify performance bottlenecks and troubleshoot problems in queries:

<!-- TOC -->

- [EXPLAIN and ANALYZE](#explain-and-analyze)
- [Enable trace logging](#enable-trace-logging)
  - [Extract a trace ID](#extract-a-trace-id)
- [Retrieve system information](#retrieve-system-information)
  - [Retrieve query instance information](#retrieve-query-instance-information)

<!-- /TOC -->
```

File: content/influxdb/clustered/query-data/execute-queries/optimize-queries.md [hunk 15]
```diff
@@ -0,0 +1,410 @@
...
+
+### EXPLAIN and ANALYZE
```
Comment (sanderson) [inline content/influxdb/clustered/query-data/execute-queries/optimize-queries.md:33]:
```suggestion
## EXPLAIN and ANALYZE
```

File: content/influxdb/clustered/query-data/execute-queries/optimize-queries.md [hunk 16]
```diff
@@ -0,0 +1,410 @@
...
+
+To view the query engine's execution plan and metrics for an SQL query, prepend [`EXPLAIN`](/influxdb/clustered/reference/sql/explain/) or [`EXPLAIN ANALYZE`](/influxdb/clustered/reference/sql/explain/#explain-analyze) to the query.
```
Comment (sanderson) [inline content/influxdb/clustered/query-data/execute-queries/optimize-queries.md:35]:
```suggestion
To view the query engine's execution plan and metrics for an SQL or InfluxQL query, prepend [`EXPLAIN`](/influxdb/clustered/reference/sql/explain/) or [`EXPLAIN ANALYZE`](/influxdb/clustered/reference/sql/explain/#explain-analyze) to the query.
```

File: content/influxdb/clustered/query-data/execute-queries/optimize-queries.md [hunk 17]
```diff
@@ -0,0 +1,410 @@
...
+|:------------------|:---------------------------------------------------------------------------------------------------------------------------------------------|
+| logical_plan      | Sort: home.time ASC NULLS LAST                                                                                                               |
+|                   |   TableScan: home projection=[co, hum, room, sensor, temp, time], full_filters=[home.time >= TimestampNanosecond(1688491380936276013, None)] |
+| physical_plan     | SortExec: expr=[time@5 ASC NULLS LAST]                                                                                                       |
+|                   |   EmptyExec: produce_one_row=false                                                                                                           |
+| Plan with Metrics | SortExec: expr=[time@5 ASC NULLS LAST], metrics=[output_rows=0, elapsed_compute=1ns, spill_count=0, spilled_bytes=0]                         |
+|                   |   EmptyExec: produce_one_row=false, metrics=[]
+```
+
+### Enable trace logging
```
Comment (sanderson) [inline content/influxdb/clustered/query-data/execute-queries/optimize-queries.md:105]:
```suggestion
## Enable trace logging
```

File: content/influxdb/clustered/query-data/execute-queries/optimize-queries.md [hunk 18]
```diff
@@ -0,0 +1,410 @@
...
+
+Replace the following:
+
+- {{% code-placeholder-key %}}`DATABASE_NAME`{{% /code-placeholder-key %}}: your {{% product-name %}} database
+- {{% code-placeholder-key %}}`DATABASE_TOKEN`{{% /code-placeholder-key %}}: a [database token](/influxdb/clustered/admin/tokens/) with sufficient permissions to the specified database
+- {{% code-placeholder-key %}}`APP_REQUEST_ID`{{% /code-placeholder-key %}}: a unique identifier you provide to associate with the trace logging.
+
+With tracing enabled, the response contains the `trace-id` header with the 128-bit unique _trace ID_ generated by the tracing system.
+
+#### Extract a trace ID
```
Comment (sanderson) [inline content/influxdb/clustered/query-data/execute-queries/optimize-queries.md:223]:
```suggestion
### Extract a trace ID
```

File: content/influxdb/clustered/query-data/execute-queries/optimize-queries.md [hunk 19]
```diff
@@ -0,0 +1,410 @@
...
+
+  # Access the trace ID that the the middleware collected.
+  assert res.traces[0]
+  print(f"Received generated trace-id: {res.traces[0]}")
+
+query_with_middleware()
+```
+{{% /code-placeholders %}}
+
+### Retrieve system information
```
Comment (sanderson) [inline content/influxdb/clustered/query-data/execute-queries/optimize-queries.md:300]:
```suggestion
## Retrieve system information
```

File: content/influxdb/clustered/query-data/execute-queries/optimize-queries.md [hunk 20]
```diff
@@ -0,0 +1,410 @@
...
+
+In addition to the SQL standard `information_schema`, {{% product-name %}} contains _system_ tables that provide access to
+InfluxDB-specific information.
+The information in each system table is scoped to the namespace you're querying;
+you can only retrieve system information for that particular instance.
+
+#### Retrieve query instance information
```
Comment (sanderson) [inline content/influxdb/clustered/query-data/execute-queries/optimize-queries.md:307]:
```suggestion
### Retrieve query instance information
```

File: content/influxdb/clustered/query-data/execute-queries/optimize-queries.md [hunk 21]
```diff
@@ -0,0 +1,410 @@
...
+
+To get information about queries you've run on the current instance, use SQL to query the [`system.queries` table](/influxdb/clustered/reference/internals/system-tables/#systemqueries-measurement).
+`system.queries` contains information about queries run against the IOx instance that is currently handling queries.
+If you [enabled trace logging for the query](#enable-trace-logging-for-a-query), the `trace-id` appears in the `system.queries.trace_id` column for the query run.
+
+The `system.queries` table is an InfluxDB v3 **debug feature**.
+To enable the feature and query `system.queries`, include an `"iox-debug"` header set to `"true"` and use SQL to query the table
```
Comment (sanderson) [inline content/influxdb/clustered/query-data/execute-queries/optimize-queries.md:314]:
```suggestion
To enable the feature and query `system.queries`, include an `"iox-debug"` header set to `"true"` and use SQL to query the table.
```

**Review by sanderson — CHANGES_REQUESTED** [review]
Looks good. Just some minor suggestions.

File: content/influxdb/clustered/query-data/execute-queries/optimize-queries.md [hunk 22]
```diff
@@ -0,0 +1,410 @@
...
+```
+
+### Enable trace logging
+
+Use query tracing in {{% product-name %}} to get observability into a slow query.
+The tracing system follows the [OpenTelemetry traces](https://opentelemetry.io/docs/concepts/signals/traces/) model for providing observability into a request.
+
+To enable tracing for your query, include the `influx-debug-id` header and a unique string value that identifies your query.
+With tracing enabled, InfluxDB generates a _trace ID_ that uniquely identifies the trace and propagates through the system.
+InfluxDB Support can then use the ID you provide and the associated trace ID to filter and correlate related log entries for the query.
```
Comment (btasker) [inline content/influxdb/clustered/query-data/execute-queries/optimize-queries.md:112]: This probably needs tweaking for Clustered. I *assume* Clustered still has the bits for tracing, but it wouldn't be Support retrieving it (unless the traces are being sent offsite?).

If Clustered definitely supports tracing, the wording is probably "InfluxDB Support can then help you use the ID..."

File: content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md [hunk 23]
```diff
@@ -0,0 +1,410 @@
...
+| physical_plan     | SortExec: expr=[time@5 ASC NULLS LAST]                                                                                                       |
+|                   |   EmptyExec: produce_one_row=false                                                                                                           |
+| Plan with Metrics | SortExec: expr=[time@5 ASC NULLS LAST], metrics=[output_rows=0, elapsed_compute=1ns, spill_count=0, spilled_bytes=0]                         |
+|                   |   EmptyExec: produce_one_row=false, metrics=[]
+```
+
+### Enable trace logging
+
+Use query tracing in {{% product-name %}} to get observability into a slow query.
+The tracing system follows the [OpenTelemetry traces](https://opentelemetry.io/docs/concepts/signals/traces/) model for providing observability into a request.
```
Comment (btasker) [inline content/influxdb/cloud-dedicated/query-data/execute-queries/optimize-queries.md:108]: Probably worth noting somewhere that adding trace headers to all queries, all the time, is not recommended.

**Review by btasker — APPROVED** [review]

File: content/influxdb/cloud-serverless/query-data/execute-queries/optimize-queries.md [hunk 24]
```diff
@@ -0,0 +1,106 @@
...
+
+  sql_explain_analyze = '''EXPLAIN ANALYZE SELECT *
+      FROM home
+      WHERE time >= now() - INTERVAL '90 days'
+      ORDER BY time'''
+
+  table = client.query(sql_explain_analyze)
+
+  # Combine the Dataframes and output the plan information.
+  df = pd.concat([df, table.to_pandas()])
```
Comment (crepererum) [inline content/influxdb/cloud-serverless/query-data/execute-queries/optimize-queries.md:73]: The information shown by `EXPLAIN ANALYZE` should be a superset of a plain `EXPLAIN` query. There's no need to combine both. Either use one or the other.

**Review by sanderson — APPROVED** [review]

**Pull request merged**